### PR TITLE
Fix flakiness of commands_test

### PR DESCRIPTION
**Pull request by B3rn475**
- Wait for completion of Hot Reload
- Wait for completion of Restart
- Fallback if checkHealth throws METHOD_NOT_FOUND.
  We try to wait for the service extensions to be registered and retry.

### CODE REVIEW

File: packages/flutter_driver/lib/src/driver.dart [hunk 1]
```diff
@@ -269,8 +270,27 @@ class FlutterDriver {
       );
     }
 
-    // At this point the service extension must be installed. Verify it.
-    final Health health = await driver.checkHealth();
+    // Invoked checkHealth and try to fix dalays in the registration of Service
```
Comment (cbracken) [inline packages/flutter_driver/lib/src/driver.dart]: Nit: s/dalays/delays/

File: packages/flutter_driver/lib/src/driver.dart [hunk 2]
```diff
@@ -269,8 +270,27 @@ class FlutterDriver {
...
+    // extensions
+    Future<Health> checkHealth() async {
+      try {
+        // At this point the service extension must be installed. Verify it.
+        return await driver.checkHealth();
```
Comment (cbracken) [inline packages/flutter_driver/lib/src/driver.dart]: Why not just `return driver.checkHealth()`?

File: packages/flutter_driver/lib/src/driver.dart [hunk 3]
```diff
@@ -8,6 +8,7 @@ import 'dart:io';
 
 import 'package:file/file.dart' as f;
 import 'package:json_rpc_2/json_rpc_2.dart' as rpc;
+import 'package:json_rpc_2/error_code.dart' as error_code;
```
Comment (cbracken) [inline packages/flutter_driver/lib/src/driver.dart]: nit: move this up one line above json_rpc_2.dart

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

File: packages/flutter_driver/lib/src/driver.dart (hunk 3 above)
Comment (B3rn475) [inline packages/flutter_driver/lib/src/driver.dart]: done

File: packages/flutter_driver/lib/src/driver.dart (hunk 1 above)
Comment (B3rn475) [inline packages/flutter_driver/lib/src/driver.dart]: done

File: dev/devicelab/manifest.yaml [hunk 4]
```diff
@@ -125,7 +125,6 @@ tasks:
       Runs tests of flutter run commands.
     stage: devicelab
     required_agent_capabilities: ["has-android-device"]
-    flaky: true
```
Comment (Hixie) [inline dev/devicelab/manifest.yaml]: that's optimistic :-)

File: packages/flutter_driver/lib/src/driver.dart [hunk 5]
```diff
@@ -207,30 +208,30 @@ class FlutterDriver {
       });
     }
 
+    // Waits for a signal from the VM service that the extension is registered
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/driver.dart]: Document what the returned string is?

File: dev/devicelab/manifest.yaml (hunk 4 above)
Comment (B3rn475) [inline dev/devicelab/manifest.yaml]: It was not failing every two trials ;-)

**Comment by Hixie** [comment]
cc @yjbanov 

LGTM

File: packages/flutter_driver/lib/src/driver.dart (hunk 5 above)
Comment (B3rn475) [inline packages/flutter_driver/lib/src/driver.dart]: I just moved this out from the inner block, but the return string should be equal to the `_kFlutterExtensionMethod` constant

**Pull request merged**